### PR TITLE
Feat: add method id/name filter for txs api

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
link: #762 
1. add method id/ method name filter for calling graphql transactions api
2. new transaction will had method id/name if transaction type is  polyjuice contracts
3. add mix task to process history polyjuice contracts's transaction

## Check List
1. check method id/name filter for graphql transactions api
2. check new polyjuice contracts transaction method id and name(if exists) was updated when import transactions to table
3. check history data when mix task finished

#### Test
- unit test
#### Task
 - need process transaction history data

### CODE REVIEW

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex [hunk 1]
```diff
@@ -230,6 +231,28 @@ defmodule GodwokenExplorer.Graphql.Resolvers.Transaction do
     end
   end
 
+  defp query_with_method_id_name({:error, _} = error, _input), do: error
+
+  defp query_with_method_id_name(query, input) do
+    method_id = Map.get(input, :method_id)
+    method_name = Map.get(input, :method_name)
+
+    query =
```
Comment (Keith-CY) [inline lib/godwoken_explorer/graphql/resolovers/transaction.ex:240]: `method_id` and `method_name` should be mutually exclusive, and `method_id` should be prior to `method_name` cuz every `method_name` has its `method_id` but not every `method_id` has its `method_name`

File: priv/repo/migrations/20221008100002_alter_transaction_with_method_id_name copy.exs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+defmodule GodwokenExplorer.Repo.Migrations.AlterTransactionWithMethodIdName do
```
Comment (Keith-CY) [inline priv/repo/migrations/20221008100002_alter_transaction_with_method_id_name copy.exs:1]: why the name has `copy` in it

File: lib/mix/tasks/update_txs_method_id_name.ex [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            {"0x0", nil}
+        end
+
+      map
+      |> Map.delete(:input)
+      |> Map.put(:method_id, method_id)
+      |> Map.put(:method_name, method_name)
+    end)
+  end
+end
```
Comment (Keith-CY) [inline lib/mix/tasks/update_txs_method_id_name.ex:145]: Maybe another task is required to iterate transactions of a specific contract once a contract is verified and abi is cached.

File: lib/mix/tasks/update_txs_method_id_name.ex [hunk 4]
```diff
@@ -0,0 +1,145 @@
+defmodule Mix.Tasks.UpdateTxsMethodIdName do
+  @moduledoc "Printed when the user requests `mix help update_txs_method_id_name`"
+
+  @shortdoc "running with limit/start/walk args `mix update_txs_method_id_name 350000 0 500` to updating udt balance and current udt balance with token_id/token_type"
```
Comment (Keith-CY) [inline lib/mix/tasks/update_txs_method_id_name.ex:4]: Why would this task update udt balance?

**Comment by zmcNotafraid** [comment]
这里还有一个可以处理的点，就是一个合约当时不存在，后来补上了，这时候怎么去更新method_id 和 method_name

File: priv/repo/migrations/20221008100002_alter_transaction_with_method_id_name copy.exs (hunk 2 above)
Comment (Naupio) [reply priv/repo/migrations/20221008100002_alter_transaction_with_method_id_name copy.exs]: fixed

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/mix/tasks/update_txs_method_id_name.ex (hunk 3 above)
Comment (zmcNotafraid) [reply lib/mix/tasks/update_txs_method_id_name.ex:140]: 这个没解决把？

File: lib/mix/tasks/update_txs_method_id_name.ex (hunk 4 above)
Comment (Naupio) [reply lib/mix/tasks/update_txs_method_id_name.ex:4]: deleted unused alias

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/transaction.ex]: fixed

File: lib/godwoken_indexer/block/sync_worker.ex [hunk 5]
```diff
@@ -493,8 +495,53 @@ defmodule GodwokenIndexer.Block.SyncWorker do
     end
   end
 
+  def add_method_id_and_name_to_tx_params(txs_params) do
+    {txs_hashes, txs_to_account_ids} =
```
[the remark this answers is not in the document]
Comment (Naupio) [reply lib/godwoken_indexer/block/sync_worker.ex:505]: added `add_method_id_and_name_to_tx_params` use with handle_polyjuice_transactions

File: lib/godwoken_indexer/block/sync_worker.ex [hunk 6]
```diff
@@ -493,8 +495,53 @@ defmodule GodwokenIndexer.Block.SyncWorker do
...
+      Enum.map(txs_params, fn tx ->
+        {:ok, r} = Hash.Full.cast(tx.hash)
+        {r, tx.to_account_id}
+      end)
+      |> Enum.unzip()
+
+    polyjuices = from(p in Polyjuice, where: p.tx_hash in ^txs_hashes) |> Repo.all()
+
+    accounts = from(a in Account, where: a.id in ^txs_to_account_ids) |> Repo.all()
```
[the remark this answers is not in the document]
Comment (Naupio) [reply lib/godwoken_indexer/block/sync_worker.ex]: fixed

File: lib/mix/tasks/update_txs_method_id_name.ex (hunk 3 above)
Comment (Naupio) [reply lib/mix/tasks/update_txs_method_id_name.ex:145]: mix task added precache for abi

**Comment by Naupio** [comment]
> 这里还有一个可以处理的点，就是一个合约当时不存在，后来补上了，这时候怎么去更新method_id 和 method_name

add this CheckUpdateTransactionMethodIdName worker to process this senario

**Review by FrederLu — APPROVED** [review]

**Comment by Naupio** [comment]
@zmcNotafraid @Keith-CY  stg was verified, pls review the code

File: test/graphql/transaction_test.exs [hunk 7]
```diff
@@ -328,6 +330,47 @@ defmodule GodwokenExplorer.Graphql.TransactionTest do
...
+    }
+    """
+
+    conn =
+      post(conn, "/graphql", %{
+        "query" => query,
+        "variables" => %{}
+      })
+
+    assert match?(
```
Comment (Keith-CY) [inline test/graphql/transaction_test.exs:357]: There should be another case that if `method id in query is not 0x12345678`, records of `method id is 0x12345678` should not be returned

Comment (Naupio) [reply test/graphql/transaction_test.exs:447]: test case was added

**Review by Keith-CY — APPROVED** [review]

**Review by zmcNotafraid — APPROVED** [review]

**Pull request merged**